code/src/layouts/header/components/header-toolbar/header-admin-toolbar/HeaderAdminToolbar.tsx

```diff
@@ -0,0 +1,54 @@
+import DashboardCustomizeIcon from "@mui/icons-material/DashboardCustomize";
+import LogoutIcon from "@mui/icons-material/Logout";
+
+import AppButton from "@/components/app-button/AppButton";
+import AppIconButton from "@/components/app-icon-button/AppIconButton";
+import AppLink from "@/components/app-link/AppLink";
+import AppTooltip from "@/components/app-tooltip/AppTooltip";
+
+import routes from "@/constants/routes";
+import { useAppDispatch } from "@/hooks/use-redux/useRedux";
+import cartApi from "@/store/api/cartApi";
+import { clearLocalCart } from "@/store/slices/localCart";
+import { logout } from "@/store/slices/userSlice";
+
+import "@/layouts/header/components/header-toolbar/HeaderToolbar.scss";
+
+const HeaderAdminToolbar = () => {
+  const dispatch = useAppDispatch();
+
+  const handleLogout = () => {
+    dispatch(logout());
+    dispatch(clearLocalCart());
+    dispatch(cartApi.util.resetApiState());
+  };
```

Comment:
I noticed that the logic of logout is repeated in toolbar components. I suggest creating component `LogoutButton` or custom hook `useLogout`

Reply:
I created both hook and component. I guess hook can be useful in other parts of our application